pyhmmer/hmmer.py

```diff
@@ -252,6 +253,43 @@ def _(self, query: DigitalMSA) -> TopHits:  # type: ignore
         return self.pipeline.search_msa(query, self.targets, self.builder)
 
 
+class _JACKHMMERWorker(_BaseWorker[_JACKHMMERQueryType, typing.Union[DigitalSequenceBlock, SequenceFile]]):
+    
+    def __init__(
+        self,
+        max_iterations: typing.Optional[int] = 5,
+        select_hits: typing.Optional[typing.Callable[[TopHits], None]] = None,
+        *args,
+        **kwargs
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.select_hits = select_hits
+        self.max_iterations = max_iterations
+
+    @singledispatchmethod
+    def query(self, query) -> TopHits:  # type: ignore
+        raise TypeError("Unsupported query type for `jackhmmer`: {}".format(type(query).__name__))
+
+    def _iterate(self, iterator) -> TopHits:
+        for n in range(self.max_iterations):
+            iteration = next(iterator)
+            if iteration.converged:
+                break
+        # unpack results
+        hmm, hits, msa, converged, it = iteration
+        return hits
```

Comment:
Making the decision here to just return `TopHits` to be consistent with the other top-level interfaces. I'm undecided if the entire `IterationResult` should be returned instead.

Reply:
I think it should return the last `IterationResult` here, because it's not impossible that the end-user wants to recover the MSA instead of the hits. I'm also wondering whether it could be worth returning more than the last iteration, but maybe this could need to much memory.

Reply:
Sounds good on both points... 

hmmer3's jackhmmer has the option to save alignment and HMM checkpoints to file, so maybe we also leave it up to the user? But we also return all the hits for each iteration, not just the HMM and alignments.

Reply:
see updated code